### ISSUE TITLE
PymodeRun Version Support

### ISSUE THREAD

**Issue by AceofSpades5757** [python-mode/python-mode#1107]
I've been trying to figure out for a while now whether or not the command supports different versions of Python, particularly in conjunction with a virtual environment. I've been thoroughly unable to find any information on the topic and have been unable to solve the issue. The only version I can seem to use is the one run in conjunction with Vim.

I think I've thoroughly read the README, FAQ, and looked through the `run` scripts, but I wasn't able to find anything.

_Please let me know if any additional information would be useful._

**Comment by lieryan** [comment]
python-mode always runs on the Python built into Vim. It can load virtualenv, and can do so even if the Vim's Python version differs from the virtualenv's Python version. It'll use the virtualenv directory to search for libraries but run it with the Vim's Python interpreter. Usually this works fine, but can sometimes be a problem if Vim's Python is incompatible with the libraries installed in the virtualenv.

You have to recompile Vim if you want to use a different Python version in Vim. If you want to experiment with different Python versions, I'd suggest using NeoVim where you can switch the Python interpreter by setting the `g:python_host_prog`/`g:python3_host_prog` variable.

**Comment by AceofSpades5757** [comment]
Thank you very much for your response and the information.

I was actually running into that issue at one point where my Vim's Python version was incompatible with something in my virtual environment. After compiling a new version, python-mode was giving me an error for not having Python 2 and then gave me another error when I compiled it with support for Python 2. It's probably something I'm doing wrong during the compilation process, but I'll worry about it some other time.

Neovim sounds interesting, but I'm not yet convinced to switch.

I actually just changed the `run_code()` function to pass the file to Python (Windows: `py -3.85 %` basically) whenever inside a virtual environment. Seems to work just fine along with the existing code, at least for the time being.
